File: Algorithms/27. Remove Element/kotlin/solution.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var i = 0
        for (j in 0 until nums.size) {
            if (nums[j] != `val`) {
                nums[i] = nums[j]
                i++
            }
        }
        return i
    }
}

File: Algorithms/27. Remove Element/swift/solution.swift
class Solution {
    func removeElement(_ nums: inout [Int], _ val: Int) -> Int {
        var count = nums.count
        var index = count - 1

        while index >= 0 {
            if nums[index] == val {
                nums.remove(at: index)
                count -= 1
            }

            index -= 1
        }

        return count
    }
}
